### PR TITLE
Improve handling edge cases

### PR DESCRIPTION
**Pull request by shankarsridhar**
This PR addresses some comments(from https://github.com/garris/BackstopJS/pull/1216) around handling some edge cases in a better way.

**Changes:**
* better handling of boolean flag - incremental flag
* stricter check for `/dtest` based on which reference or test modes are selected.

_P.S: Tested against _ember-backstop-tutorial_ repo._

### CODE REVIEW

File: remote/index.js [hunk 1]
```diff
@@ -14,6 +14,8 @@ var express = require('express');
 var backstop = require('../core/runner');
 var { modifyJsonpReport } = require('../core/util/remote');
 
+const booleanizeArg = incrementalFlag => [true, 'true'].includes(incrementalFlag);
```
Comment (garris) [inline remote/index.js:17]: What happens when incrementalFlag is `"false"`?

File: remote/index.js [hunk 2]
```diff
@@ -66,8 +68,8 @@ module.exports = function (app) {
       vid: app._backstop.testCtr
     };
 
-    const command = req.path.includes('dtest') ? 'test' : 'reference';
-    backstop(command, { config, i: Boolean(req.body.i) }).then(
+    const command = req.path.includes(`/dtest/`) ? 'test' : 'reference';
```
Comment (garris) [inline remote/index.js:71]: Would you mind instead checking for `/dref/` so that default behavior is `test`?  I know it's not currently a concern, but we may alias more routes to this flow.

File: remote/index.js (hunk 1 above)
Comment (shankarsridhar) [reply remote/index.js:17]: This method(`booleanizeArg()`) will return boolean `false` when the input  is anything other than `true` or `'true'`.

File: remote/index.js (hunk 2 above)
Comment (shankarsridhar) [reply remote/index.js]: I've updated the PR accordingly.

File: remote/index.js (hunk 1 above)
Comment (garris) [reply remote/index.js:17]: awesome.
